mermithid/processors/misc/ComplexLineShape.py

```diff
@@ -0,0 +1,598 @@
+m_e = 9.10938356*10**(-31) # Kilograms , mass of electron
+mass_energy_electron = 510.9989461 # keV
+
+# A lorentzian function
+def lorentzian(x_array,x0,FWHM):
+    HWHM = FWHM/2.
+    func = (1./np.pi)*HWHM/((x_array-x0)**2.+HWHM**2.)
+    return func
+
+# A gaussian function
+def gaussian(x_array,A,sigma,mu):
+    f = A*(1./(sigma*np.sqrt(2*np.pi)))*np.exp(-(((x_array-mu)/sigma)**2.)/2.)
+    return f
+
+# Converts a gaussian's FWHM to sigma
+def gaussian_FWHM_to_sigma(FWHM):
+    sigma = FWHM/(2*np.sqrt(2*np.log(2)))
+    return sigma
+
+# Converts a gaussian's sigma to FWHM
+def gaussian_sigma_to_FWHM(sigma):
+    FWHM = sigma*(2*np.sqrt(2*np.log(2)))
+    return FWHM
+
+# calculating combinatorics n choose r
+def ncr(n, r):
+    r = min(r, n-r)
+    if r == 0: return 1
+    res = 1
+    for k in range(1,r+1):
+        res = res*(n-k+1)/k
+    return res
+
+#returns array with energy loss/ oscillator strength data
+def read_oscillator_str_file(filename):
+    f = open(filename, "r")
+    lines = f.readlines()
+    f.close()
+    energyOsc = [[],[]] #2d array of energy losses, oscillator strengths
+
+    for line in lines:
+        if line != "" and line[0]!="#":
+            raw_data = [float(i) for i in line.split("\t")]
+            energyOsc[0].append(raw_data[0])
+            energyOsc[1].append(raw_data[1])
+
+    energyOsc = np.array(energyOsc)
+    ### take data and sort by energy
+    sorted_indices = energyOsc[0].argsort()
+    energyOsc[0] = energyOsc[0][sorted_indices]
+    energyOsc[1] = energyOsc[1][sorted_indices]
+    return energyOsc
+
+# A sub function for the scatter function. Found in
+# "Energy loss of 18 keV electrons in gaseous T and quench condensed D films"
+# by V.N. Aseev et al. 2000
+def aseev_func_tail(energy_loss_array, gas_type):
+    if gas_type=="H2":
+        A2, omeg2, eps2 = 0.195, 14.13, 10.60
+    elif gas_type=="Kr":
+        A2, omeg2, eps2 = 0.4019, 22.31, 16.725
+    return A2*omeg2**2./(omeg2**2.+4*(energy_loss_array-eps2)**2.)
+
+#convert oscillator strength into energy loss spectrum
+def get_eloss_spec(e_loss, oscillator_strength, kr_line): #energies in eV
+    kinetic_en = kr_line * 1000
+    e_rydberg = 13.605693009 #rydberg energy (eV)
+    a0 = 5.291772e-11 #bohr radius
+    return np.where(e_loss>0 , 4.*np.pi*a0**2 * e_rydberg / (kinetic_en * e_loss) * oscillator_strength * np.log(4. * kinetic_en * e_loss / (e_rydberg**3.) ), 0)
+
+# Takes only the nonzero bins of a histogram
+def get_only_nonzero_bins(bins,hist):
+    nonzero_idx = np.where(hist!=0)
+    hist_nonzero = hist[nonzero_idx]
+    hist_err = np.sqrt(hist_nonzero)
+    bins_nonzero = bins[nonzero_idx]
+    return bins_nonzero , hist_nonzero , hist_err
+
+# Flips an array left-to-right. Useful for converting between energy and frequency
+def flip_array(array):
+    flipped = np.fliplr([array]).flatten()
+    return flipped
+
+# Given energy in keV and the self.B_field of the trap, returns frequency in Hz
```

Comment:
this function is already defined in other places. once we merge the efficiency-weights pr we could merge the phase-2-analysis branch into this branch. this function will then exists in misc/ConversionFunctions.py and could be imported.